tensorflow-framework/src/main/java/org/tensorflow/framework/losses/impl/LossesImpl.java

```diff
@@ -0,0 +1,304 @@
+package org.tensorflow.framework.losses.impl;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.losses.Reduction;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.core.ReduceSum;
+import org.tensorflow.op.core.Squeeze;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.types.family.TNumber;
+
+import java.util.Collections;
+
+public class LossesImpl {
+
+  /**
+   * Squeeze or expand last dimension if needed with a sampleWeights of one.
+   *
+   * <ol type="1">
+   *   <li>Squeezes last dim of <code>predictions</code> or <code>labels</code> if their rank differs by 1 (using
+   *       {@link #removeSqueezableDimensions}).
+   *   <li>Squeezes or expands last dim of <code>sampleWeight` if its rank differs by 1 from the new
+   *       rank of <code>predictions`. If <code>sampleWeight` is scalar, it is kept scalar./li>
+   * </ol>
+   *
+   * @param tf the TensorFlow Ops
+   * @param predictions Predicted values, a <code>Operand</code> of arbitrary dimensions.
+   * @param labels Optional label <code>Operand</code> whose dimensions match <code>prediction</code>.
+   * @return Tuple of <code>prediction</code>, <code>label</code> and <code>sampleWeight</code>. Each of them possibly has the last
```

Comment:
For this method, the returned `sampleWeight` is always null.

Reply:
That is not always the case when we do Metrics.

Reply:
I'm just thinking our documentation for this method might take into account that the returned `sampleWeight` is always `null`.

Reply:
Now I see what you  are talking about. I added a comment in the @return that `sampleWeight` will be `null` for this particular method signature.